src/app/features/terms-of-use/terms-of-use.page.ts

```diff
@@ -0,0 +1,45 @@
+import { Component } from '@angular/core';
+import { NavController } from '@ionic/angular';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import { BehaviorSubject, fromEvent } from 'rxjs';
+import { tap } from 'rxjs/operators';
+import { BUBBLE_IFRAME_URL } from '../../shared/dia-backend/secret';
+import { NetworkService } from '../../shared/network/network.service';
+
+@UntilDestroy()
+@Component({
+  selector: 'app-terms-of-use',
+  templateUrl: './terms-of-use.page.html',
+  styleUrls: ['./terms-of-use.page.scss'],
+})
+export class TermsOfUsePage {
+  readonly networkConnected$ = this.networkService.connected$;
+  readonly termsOfUseUrl = `${BUBBLE_IFRAME_URL}/terms_of_use`;
+  readonly iframeLoaded$ = new BehaviorSubject(false);
+
+  constructor(
+    private readonly networkService: NetworkService,
+    private readonly navController: NavController
+  ) {}
+
+  ionViewDidEnter() {
+    this.processIframeEvents();
+  }
+
+  processIframeEvents() {
+    fromEvent(window, 'message')
+      .pipe(
+        tap(event => {
+          const postMessageEvent = event as MessageEvent;
+          if (postMessageEvent.data === 'iframe-on-load') {
+            this.iframeLoaded$.next(true);
+          }
+          if (postMessageEvent.data === 'iframeBackButtonClicked') {
+            this.navController.back();
+          }
+        }),
+        untilDestroyed(this)
+      )
+      .subscribe();
+  }
+}
```

Comment:
`ChatGPT code review for git diff 6c6d81d3..45109709 "*terms-of-use.page.ts"`

This code defines a `TermsOfUsePage` component for an Ionic application. The component displays the terms of use in an iframe, and handles events coming from the iframe. It listens to the network connection status and iframe's load event by using observables, and it also listens to the iframe's back button event by using event listener.

Improvement suggestions:

-   Extract the common functionality of the two pages `DataPolicyPage` and `TermsOfUsePage` into a shared service or a base page class to avoid duplicated code
-   Instead of using a magic string "iframe-on-load", you can define a constant variable for this string value, this can make your code more readable and maintainable
-   The `iframeLoaded$` variable should be defined as a private variable and not exposed as a public variable.
-   Use `const` keyword instead of `let` for variables that are not reassigned.
-   Make sure that the URL used to load the iframe is a trusted source, as using untrusted URLs can lead to security vulnerabilities such as XSS attacks.

Reply:
@tammyyang regarding improvement suggestions:

- Make sense, but it's better when things are repeated more than 2 times otherwise it's the same amount of code. Would ignore this suggestion.
- Make sense, related asana cards are created.
  - [[Code Quality + Security] capture-app-ionic. Unify iframe postMessage format and extract to ENUMs](https://app.asana.com/0/1201016280880500/1203794276110745/f)
  - [[Code Quality + Security] capture-app-iframe. Unify iframe postMessage format](https://app.asana.com/0/0/1203794276110749/f)
- Agree but maybe we can keep it public.
- For this commit for variables that are not reassigned I use `const` and `readonly` already.
- iframe URL is assigned at the CI build phase. Therefore URL for the iframe is highly likely to be safe

Reply:
Arrange suggestion tasks in the next sprint